subsys/bluetooth/host/gatt.c

```diff
@@ -864,7 +864,9 @@ int bt_gatt_service_register(struct bt_gatt_service *svc)
 	sc_indicate(&gatt_sc, svc->attrs[0].handle,
 		    svc->attrs[svc->attr_count - 1].handle);
 
-	db_changed();
+	if (IS_ENABLED(CONFIG_BT_GATT_CACHING)) {
+		db_changed();
```

Comment:
There's a second call in `bt_gatt_service_unregister()`. Shouldn't you change that too? And when you do, I suppose you can remove the #ifdef from inside the `db_changed()` function. If you don't remove the #ifdef from the function then I suppose this kind of changes don't do much since the call will either way be optimised away.

Reply:
Actually this one I couldn't make it work because the db_changed has dependencies.